src/video/canvas/texture.js

```diff
@@ -140,12 +140,22 @@
                     // Source coordinates
                     var s = frame.frame;
 
+                    var originX, originY;
+                    // Pixel-based offset origin from the top-left of the source frame
+                    var hasTextureAnchorPoint = (frame.spriteSourceSize && frame.sourceSize && frame.pivot);
+                    if (hasTextureAnchorPoint) {
+                        originX = (frame.sourceSize.w * frame.pivot.x) - frame.spriteSourceSize.x;
+                        originY = (frame.sourceSize.h * frame.pivot.y) - frame.spriteSourceSize.y;
+                    }
+                    
                     atlas[frame.filename] = {
-                        name    : name, // frame name
-                        offset  : new me.Vector2d(s.x, s.y),
-                        width   : s.w,
-                        height  : s.h,
-                        angle   : (frame.rotated === true) ? nhPI : 0
+                        name         : name, // frame name
+                        offset       : new me.Vector2d(s.x, s.y),
+                        hasTextureAnchorPoint : hasTextureAnchorPoint,
+                        anchorPoint  : (hasTextureAnchorPoint) ? new me.Vector2d(originX / s.w, originY / s.h) : null,
```

Comment:
The problem that came up in my head was where one frame in an animation had an anchorPoint defined but another did not. But looking back having a separate boolean didn't exactly fix that problem. I'm not setting the original anchorPoint back to default again, so that problem isn't entirely fixed (rare case, though)